kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtComponentServiceImpl.java

```diff
@@ -244,12 +250,20 @@ public GwtConfigComponent findWireComponentConfigurationFromPid(GwtXSRFToken xsr
                 if (conf != null) {
                     conf.getConfigurationProperties().put(ConfigurationAdmin.SERVICE_FACTORYPID, factoryPid);
                 }
-                if ((conf != null) && (conf.getDefinition() == null)) {
+                if (conf != null && conf.getDefinition() == null) {
                     String temporaryName = String.valueOf(System.nanoTime());
                     cs.createFactoryConfiguration(factoryPid, temporaryName, extraProps, false);
                     try {
-                        // wait for the services to be up
-                        TimeUnit.MILLISECONDS.sleep(500);
+
```

Comment:
Having a thread sleep, in this case, does not guarantee that the service will be up. This is especially true for low powered devices.
I'm not sure that the waitForService will be executed in a different thread. Looking at the code [here](http://grepcode.com/file/repo1.maven.org/maven2/org.apache.felix/org.apache.felix.framework/4.6.1/org/osgi/util/tracker/ServiceTracker.java#ServiceTracker.waitForService%28long%29), it should work as expected. Furthermore, the following line after the waitForService is a close and looking at the [javadoc](https://osgi.org/javadoc/r4v42/org/osgi/util/tracker/ServiceTracker.html#waitForService(long)) a close to the ServiceTracker, will determine the waitForService to return.